### PR TITLE
[HINTS]Add implementation for hints & utils of functions verify_zero and reduce (cairo_secp/field.cairo )

### PR DESCRIPTION
**Pull request by pefontana**
# [HINTS]Add implementation for hints & utils of functions verify_zero and reduce (cairo_secp/field.cairo )

## Description

* Add Secp::SecpUtils::pack function. 
* Hints implementation:
```
verify_zero:
%{
    from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack

    q, r = divmod(pack(ids.val, PRIME), SECP_P)
    assert r == 0, f"verify_zero: Invalid input {ids.val.d0, ids.val.d1, ids.val.d2}."
    ids.q = q % PRIME
%}
```

```
reduce:
%{
    from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack

    value = pack(ids.x, PRIME) % SECP_P
%}
```

Description of the pull request changes and motivation.

## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [x] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/vm/hints/secp/secp_utils.rs [hunk 1]
```diff
@@ -25,6 +26,24 @@ pub fn split(integer: &BigInt) -> Result<Vec<BigInt>, VirtualMachineError> {
...
+*/
+pub fn pack(d0: &BigInt, d1: &BigInt, d2: &BigInt, prime: &BigInt) -> BigInt {
+    let unreduced_big_int_3 = vec![d0, d1, d2];
+    let base: BigInt = bigint!(1) << 86_usize;
+
+    unreduced_big_int_3
+        .iter()
+        .enumerate()
+        .map(|(idx, value)| as_int(value, prime) * base.pow(idx as u32))
+        .sum()
```
Comment (Oppen) [inline src/vm/hints/secp/secp_utils.rs:44]: `(x^n)^m == x^(n*m)` so `(2⁸⁶)^n == 2^(86*n) == 2 << (86*n)`:
```suggestion
    let unreduced_big_int_3 = vec![d0, d1, d2];

    unreduced_big_int_3
        .iter()
        .enumerate()
        .map(|(idx, value)| as_int(value, prime) * (bigint!(1) << (idx * 86)))
        .sum()
```

File: src/vm/hints/secp/field_utils.rs [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+    from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+
+    q, r = divmod(pack(ids.val, PRIME), SECP_P)
+    assert r == 0, f"verify_zero: Invalid input {ids.val.d0, ids.val.d1, ids.val.d2}."
+    ids.q = q % PRIME
+%}
+*/
+pub fn verify_zero(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
```
Comment (Oppen) [inline src/vm/hints/secp/field_utils.rs:30]:
```suggestion
    ids: &HashMap<String, BigInt>,
```

File: src/vm/hints/secp/field_utils.rs [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+Implements hint:
+%{
+    from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack
+
+    value = pack(ids.x, PRIME) % SECP_P
+%}
+*/
+pub fn reduce(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
```
Comment (Oppen) [inline src/vm/hints/secp/field_utils.rs:72]:
```suggestion
    ids: &HashMap<String, BigInt>,
```

**Review by Oppen — CHANGES_REQUESTED** [review]

File: src/vm/hints/secp/secp_utils.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pefontana) [reply src/vm/hints/secp/secp_utils.rs]: Done!

**Review by Oppen — APPROVED** [review]

**Pull request merged**